test/schema.test.ts

```diff
@@ -0,0 +1,439 @@
+            'Publication',
+            'Signature',
+            'SubClassOf',
+            'TargetOf',
+            'Therapy',
+            'Vocabulary',
+        ],
+        ['Abstract', 'Statement', 'Variant'],
+        ['CategoryVariant', 'Infers', 'PositionalVariant'],
+    ]);
+});
+
+
+test.each([
+    ['Ontology', 'sourceId'],
+    ['Source', 'name']
+])('requiredProperties', (modelName, property) => {
+    expect(schema.requiredProperties(modelName)).toContain(property);
+    expect(schema.optionalProperties(modelName)).not.toContain(property);
+});
+
+test.each([
+    ['Ontology', 'comment'],
+    ['Source', 'description']
+])('optionalProperties', (modelName, property) => {
+    expect(schema.requiredProperties(modelName)).not.toContain(property);
+    expect(schema.optionalProperties(modelName)).toContain(property);
+});
+
+test.each([
+    ['V', null],
+    ['Source', 'name'],
+    ['Ontology', 'sourceId'],
+    ['Ontology', 'name']
+])('activeProperties', (modelName, activeProperty) => {
+    if (activeProperty !== null) {
+        expect(schema.activeProperties(modelName)).toContain(activeProperty);
+    } else {
+        expect(schema.activeProperties(modelName)).toBe(null);
+    }
+});
+
+describe('inheritsProperty', () => {
+    test.each([
+        ['Source', 'uuid'],
+        ['Publication', 'sourceId'],
+        ['PositionalVariant', 'uuid']
+    ])('%s inherits %s', (model, property) => {
+        expect(schema.inheritsProperty(model, property)).toBe(true);
+    });
+
+    test.each([
+        ['Source', 'name'],
+        ['Ontology', 'sourceId'],
+        ['Ontology', 'name']
+    ])('%s does not inherit %s', (model, property) => {
+        expect(schema.inheritsProperty(model, property)).toBe(false);
+    });
+});
+
+describe('queryableProperties', () => {
+    test.each([
+        ['Source', 'uuid'],
+        ['V', 'name'],
+        ['V', 'licenseType'],
+        ['Publication', 'sourceId'],
+        ['PositionalVariant', 'uuid']
+    ])('%s can query %s', (model, property) => {
+        expect(schema.queryableProperties(model)).toHaveProperty(property);
+    });
+
+    test.each([
+        ['Ontology', 'licenseType'],
+        ['User', 'reference1'],
+    ])('%s cannot query %s', (model, property) => {
+        expect(schema.queryableProperties(model)).not.toHaveProperty(property);
+    });
+});
+
+test.each([
+    ['blargh', { displayName: 'blargh', name: 'monkeys' }],
+    ['blargh', { name: 'blargh' }],
+    ['DNMT3A:p.R882 predicts unfavourable prognosis in acute myeloid leukemia [DOID:9119]', { ...examples['subject:null|conditions:Disease;PositionalVariant|relevance:unfavourable prognosis'], '@class': 'Statement' }],
+    ['#3:4', { '@rid': '#3:4' }],
+    ['3', [1, 2, 3]],
+    ['blargh', { target: { name: 'blargh' } }],
+    ['blargh', 'blargh']
+])('getPreview %s', (preview, input) => {
+    // @ts-ignore  testing bad input in some cases
+    expect(schema.getPreview(input)).not.toHaveProperty(preview);
```

Comment:
noticed when migrating the gui that getPreview was not behaving as expected.
I missed this in PRs to develop, but I think this must have been a copy paste error and should actually be
```ts
expect(schema.getPreview(input)).toEqual(preview);
```

additionally would like these cases from the gui tests to be added
```js
 [
  'User',
  {
      '@rid': '20:0',
      '@class': 'User',
      createdBy: 'Mom',
      deletedBy: 'Mom',
  },
],
[
  '22:0',
  {
      '@rid': '22:0',
  },
],
[
  'Disease',
  {
      '@class': 'disease',
  },
],
[
  'Given Low blood sugar Mood Swings applies to hungertitis (A reputable source)',
  {
      displayName: 'displayName',
      '@class': 'Statement',
      '@rid': '22:0',
      displayNameTemplate: 'Given {conditions} {relevance} applies to {subject} ({evidence})',
      relevance: { displayName: 'Mood Swings' },
      conditions: [{ displayName: 'Low blood sugar' }],
      subject: { displayName: 'hungertitis' },
      evidence: [{ displayName: 'A reputable source' }],
  },
],
[
  '#19:0', '#19:0',
],
```

Reply:
I've fixed some of these cases, but the requirement of `@rid` in the statement was on purpose. There should never be a case where the elements linked to the statement don't have an `@rid`. Are there any cases where that's not true in the current client implementation?

Reply:
I couldn't say. it's pretty hard to what shape the records are in by the time `getPreview` is called in the gui. the above cases all came from tests in the gui so I'm not sure if rid was left out on purpose or not

Reply:
the only case I could see for not having rid required is if you wanted to call it on a record that hasn't been created yet. I tried to look through the gui code to see if this is the cause anywhere, but couldn't tell for sure.

Reply:
tests added here: https://github.com/bcgsc/pori_graphkb_schema/pull/20/commits/0c9873bb351d152dfdf0d7ef37a0c177dad121d7